src/directives/ng-file-drop.ts

```diff
@@ -122,6 +122,16 @@ export class NgFileDropDirective implements OnChanges, OnInit {
       });
     }
 
+    if (this.options.maxSize > 0) {
+      this.files = this.files.filter(f => {
```

Comment:
shouldn't here be also `[].filter.call(this.files, (f: File) => ...`. I think this does not work in Firefox.